gdal vector set-geom-type: implement --auto (#14873)
OSGeo/gdal: doc/source/programs/gdal_vector_set_geom_type.rst

Program-Specific Options
------------------------

.. option:: --auto

   .. versionadded:: 3.14

>    Guess the geometry type by inspecting all input feature geometries. The most

Reviewer: ```suggestion
   Automatically determine the geometry type by inspecting all input feature geometries. The most
```

>    restrictive geometry type will be selected. For example. if there is a mix of

Reviewer: ```suggestion
   restrictive geometry type will be selected. For example, if there is a mix of
```

   polygons and multipolygons, the result will be multipolygons. If a common

>    geometry type cannot be determined, the generic Geometry type will be selected.

Reviewer: ```suggestion
   Guess the geometry type by inspecting all input feature geometries. The most
   restrictive geometry type will be selected. For example. if there is a mix of
   polygons and multipolygons, the result will be multipolygons. If a common
   geometry type cannot be determined, the generic Geometry type will be selected.
```

   .. code-block:: bash

        $ gdal vector set-geom-type --linear in.gpkg out.shp --overwrite

.. example::

>    :title: Convert a CSV file that contains only a single type of geometry into a properly typed GeooPackage

Reviewer: ```suggestion
   :title: Convert a CSV file that contains only a single type of geometry into a properly typed GeoPackage
```

>    :title: Convert a CSV files that contains only a single type of geometry into a properly typed GeooPackage

Reviewer: ```suggestion
   :title: Convert a CSV file that contains only a single type of geometry into a properly typed GeooPackage
```
